src/mpl2/src/hier_rtlmp.cpp

```diff
@@ -1123,10 +1139,71 @@ void HierRTLMP::mergeClusters(std::vector<Cluster*>& candidate_clusters)
         }
       }
     }
-    // If no more clusters have been merged, exit the merging loop
-    if (num_candidate_clusters == new_candidate_clusters.size()) {
-      break;
+    */
+
+    // Then we perform Type 2 merge
+    std::vector<Cluster*> new_candidate_clusters;
+    for (int i = 0; i < num_candidate_clusters; i++) {
+      if (cluster_class[i] == -1) {  // the cluster has not been merged
+        // new_candidate_clusters.push_back(candidate_clusters[i]);
+        for (int j = i + 1; j < num_candidate_clusters; j++) {
+          if (cluster_class[j] != -1) {
+            continue;
+          }
+          bool flag = candidate_clusters[i]->isSameConnSignature(
+              *candidate_clusters[j], signature_net_threshold_);
+          if (flag) {
+            cluster_class[j] = i;
+            bool delete_flag = false;
+            if (candidate_clusters[i]->mergeCluster(*candidate_clusters[j],
+                                                    delete_flag)) {
+              if (delete_flag) {
+                cluster_map_.erase(candidate_clusters[j]->getId());
+                delete candidate_clusters[j];
+              }
+              setInstProperty(candidate_clusters[i]);
+              setClusterMetric(candidate_clusters[i]);
+            }
+          }
+        }
+      }
+    }
+
+    // Then we perform Type 3 merge:  merge all dust cluster
```

Comment:
what is a 'dust cluster'?

Reply:
dust clusters are glue logic clusters, that contain very few instances

Reply:
Then just say that in the comment